nixos/modules/services/networking/webhook.nix

```diff
@@ -0,0 +1,189 @@
+            execute-command = "/var/scripts/redeploy.sh";
+            working-directory = "/var/webhook";
+          };
+        };
+        type = types.attrsOf (types.submodule ({ config, ... }: {
+          options = {
+            id = mkOption {
+              type = types.str;
+              default = config._module.args.name;
+              description = mdDoc "specifies the ID of your hook. This value is used to create the HTTP endpoint (http://yourserver:port/hooks/your-hook-id)";
+            };
+            execute-command = mkOption {
+              type = types.str;
+              description = mdDoc "specifies the command that should be executed when the hook is triggered";
+            };
+            working-directory = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = mdDoc "specifies the working directory that will be used for the script when it's executed";
+            };
+            response-message = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = mdDoc "specifies the string that will be returned to the hook initiator";
+            };
+            response-headers = mkOption {
+              type = types.nullOr (types.listOf
+                (types.submodule {
+                  options = {
+                    name = mkOption {
+                      type = types.str;
+                      example = "X-Example-Header";
+                      description = mdDoc "Header name";
+                    };
+                    value = mkOption {
+                      type = types.str;
+                      example = "it works";
+                      description = mdDoc "Header value";
+                    };
+                  };
+                })
+              );
+              default = null;
+              description = mdDoc "specifies the list of headers that will be returned in HTTP response for the hook";
+            };
+          };
+        }));
+      };
+
+      successHttpResponseCode = mkOption {
+        type = types.nullOr types.ints.positive;
+        default = null;
+        description = mdDoc "specifies the HTTP status code to be returned upon success";
+      };
+
+
+      triggerRule = mkOption {
+        type = types.nullOr typeTriggerRule;
+        default = null;
+        description = mdDoc "specifies the list of headers that will be returned in HTTP response for the hook";
+      };
+
+      # TODO: add other options from https://github.com/adnanh/webhook/blob/master/docs/Hook-Definition.md
+    };
+  };
+
+  imports = [];
+
+  config = mkIf cfg.enable {
+
+    users.users = mkIf (cfg.user == defaultUser) {
+      ${defaultUser} =
+        {
+          isSystemUser = true;
+          group = cfg.group;
+          description = "Webhook daemon user";
+        };
+    };
+
+    users.groups = mkIf (cfg.group == defaultUser) {
+      ${defaultUser} = {};
+    };
```

Comment:
What if only the user is set? In don't think we should create the group in that case.

Reply:
OK, changes are in https://github.com/NixOS/nixpkgs/pull/205636/commits/9b34c109c687d27efac10ac7fe75549d4c7eafa1